### DISCUSSION TITLE
warn: wayland.c:1619: compositor does not implement the XDG toplevel icon protocol

### DISCUSSION THREAD

**Discussion by charbelnicolas** [labwc/labwc#2270] [Q&A]
I've been getting this warning on my log `warn: wayland.c:1619: compositor does not implement the XDG toplevel icon protocol`, I had not seen it until today. Is this an issue with labwc?

**Comment by Consolatis** [comment]
> XDG toplevel icon protocol

That is a relatively new wayland protocol, not implemented in either labwc nor the latest wlroots release. The goal of the toplevel icon protocol is to allow applications to define window icons. However, labwc supports applications defining their window icon based on their app-id since the just released version 0.8.1.

So its an issue with labwc in the regard that we don't support the protocol yet but its in no way critical and the application in question should likely not make this a warning in the first place.
